homeassistant/components/media_player/openhome.py

```diff
@@ -0,0 +1,221 @@
+"""
+Support for Openhome Devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.openhome/
+"""
+import logging
+
+from homeassistant.components.media_player import (
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK, SUPPORT_TURN_ON,
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_SET, SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_STEP, SUPPORT_STOP, SUPPORT_PLAY, SUPPORT_SELECT_SOURCE,
+    MediaPlayerDevice)
+from homeassistant.const import (
+    STATE_IDLE, STATE_PAUSED, STATE_PLAYING, STATE_OFF)
+
+REQUIREMENTS = ['openhomedevice==0.2']
+
+SUPPORT_OPENHOME = SUPPORT_SELECT_SOURCE | \
+    SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | \
+    SUPPORT_TURN_OFF | SUPPORT_TURN_ON
+
+_LOGGER = logging.getLogger(__name__)
+
+# List of devices that have been registered
+DEVICES = []
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup Openhome Platform."""
+    from openhomedevice.Device import Device
+
+    if discovery_info:
+        _LOGGER.info('Openhome device found, (%s)', discovery_info[0])
+        device = Device(discovery_info[1])
+
+        # if device has already been discovered
+        if device.Uuid() in [x.unique_id for x in DEVICES]:
+            return True
+
+        device = OpenhomeDevice(hass, device)
+
+        add_devices([device], True)
+        DEVICES.append(device)
+
+        return True
+
+    return True
+
+
+class OpenhomeDevice(MediaPlayerDevice):
+    """Representation of an Openhome device."""
+
+    def __init__(self, hass, device):
+        """Initialise the Openhome device."""
+        self.hass = hass
+        self._device = device
+        self._track_information = {}
+        self._in_standby = None
+        self._transport_state = None
+        self._track_information = None
+        self._volume_level = None
+        self._volume_muted = None
+        self._supported_features = SUPPORT_OPENHOME
+        self._source_names = list()
+        self._name = None
+        self._state = STATE_PLAYING
+
+    def update(self):
+        """Update state of device."""
+        self._in_standby = self._device.IsInStandby()
+        self._transport_state = self._device.TransportState()
+        self._track_information = self._device.TrackInfo()
+        self._volume_level = self._device.VolumeLevel()
+        self._volume_muted = self._device.IsMuted()
+        self._source = self._device.Source()
+        self._source_index = {}
+        self._name = self._device.Room().decode('utf-8')
+        self._supported_features = SUPPORT_OPENHOME
+
+        for source in self._device.Sources():
+            self._source_names.append(source["name"])
+            self._source_index[source["name"]] = source["index"]
+
+        if self._source["type"] == "Radio":
+            if self._transport_state in ('Playing', 'Buffering'):
```

Comment:
The reason I've explicitly added `SUPPORT_PLAY` is the device also has other modes like HDMI input or Airplay in which modes play, pause and stop or meaningless, so would prefer no buttons to appear. 

This ties in with your observation about the `STATE_PLAYING` below, as on a source where transport controls are meaningless this last `else` is invoked. 

If the device has no control, but is still playing music through speakers (e.g. line-in, hdmi, phono, etc), should I report `STATE_IDLE` or `STATE_PLAYING`.